pkgs/misc/cups/drivers/mfcj6510dwlpr/default.nix

```diff
@@ -0,0 +1,71 @@
+{ stdenv, fetchurl, pkgsi686Linux, dpkg, makeWrapper, coreutils, gnused, gawk, file, cups, patchelf, utillinux, vimNox
+, ghostscript, a2ps }:
+
+stdenv.mkDerivation rec {
+  name = "mfcj6510dwlpr-${version}";
+  version = "3.0.0-1";
+
+  src = fetchurl {
+    url = "http://download.brother.com/welcome/dlf006614/mfcj6510dwlpr-${version}.i386.deb";
+    sha256 = "1ccvx393pqavsgzd8igrzlin5jrsf01d3acyvwqd1d0yz5jgqy6d";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+  buildInputs = [ cups ghostscript dpkg a2ps ];
+
+  unpackPhase = "true";
+
+  brprintconf_mfcj6510dw_script = ''
+    #!/bin/sh
+    cd $(mktemp -d)
+    ln -s @out@/usr/bin/brprintconf_mfcj6510dw_patched brprintconf_mfcj6510dw_patched
+    ln -s @out@/opt/brother/Printers/mfcj6510dw/inf/brmfcj6510dwfunc brmfcj6510dwfunc
+    ln -s @out@/opt/brother/Printers/mfcj6510dw/inf/brmfcj6510dwrc brmfcj6510dwrc
+    ./brprintconf_mfcj6510dw_patched "$@"
+  '';
+
+  installPhase = ''
+    dpkg-deb -x $src $out
+    substituteInPlace $out/opt/brother/Printers/mfcj6510dw/lpd/filtermfcj6510dw \
+      --replace /opt "$out/opt"
+    substituteInPlace $out/opt/brother/Printers/mfcj6510dw/lpd/psconvertij2 \
+      --replace "GHOST_SCRIPT=`which gs`" "GHOST_SCRIPT=${ghostscript}/bin/gs"
+    substituteInPlace $out/opt/brother/Printers/mfcj6510dw/inf/setupPrintcapij \
+      --replace "/opt/brother/Printers" "$out/opt/brother/Printers" \
+      --replace "printcap.local" "printcap"
+
+    patchelf --set-interpreter ${pkgsi686Linux.stdenv.cc.libc.out}/lib/ld-linux.so.2 \
+      --set-rpath $out/opt/brother/Printers/mfcj6510dw/inf:$out/opt/brother/Printers/mfcj6510dw/lpd \
+      $out/opt/brother/Printers/mfcj6510dw/lpd/brmfcj6510dwfilter
+    patchelf --set-interpreter ${pkgsi686Linux.stdenv.cc.libc.out}/lib/ld-linux.so.2 $out/usr/bin/brprintconf_mfcj6510dw
+
+    #stripping a hardcoded path.
+    ${utillinux}/bin/hexdump -ve '1/1 "%.2X"' $out/usr/bin/brprintconf_mfcj6510dw | \
+    sed 's.2F6F70742F62726F746865722F5072696E746572732F25732F696E662F6272257366756E63.62726d66636a36353130647766756e63000000000000000000000000000000000000000000.' | \
```

Comment:
In case the technical details are keeping people from reviewing, jump over to http://string-functions.com/hex-string.aspx and verify:
2F6F70742F62726F746865722F5072696E746572732F25732F696E662F6272257366756E63
=/opt/brother/Printers/%s/inf/br%sfunc
62726d66636a36353130647766756e63000000000000000000000000000000000000000000
= brmfcj6510dwfunc

The %s is printf's string replacement. You can run gdb and find the entry point but long story short it picks up the 10th char from the file name and inserts it there.
So, _brprintconf_mfcj6510dw_ will look for _/opt/brother/Printers/f_mfcj6510dw/inf/brf_mfcj6510dwfunc_ ... I guess they internally named the binary _printconf_mfcj6510dw_ and had it in /opt in all tested systemd.

I tried taking advantage of %s by naming the file with the nix-store hash but they limited the sampling to 20 chars AND the binary segfaults if executed in a read-only dir so that didn't workout at all. The next string in the struct\array is tightly packed and bound-checked so I couldn't make room without hunting for al the go-tos...

End result was to strip all the path and link everything in an mktemp dir on execution. It's just links so it's fairly cheap.